### DISCUSSION TITLE
Performance without roadrunner

### DISCUSSION THREAD

**Discussion by ardabeyazoglu** [spiral/spiral#836] [General]
Hi,

I know that spiral recommends roadrunner as the web server, but it is not suitable for everything especially for existing codebases. In this case, would you still recommend it with fpm ? Does spiral have any side effect on performance without roadrunner, in addition to the traditional bottlenecks that come with shared-nothing architecture ? 
I am considering migrating a legacy code base to symfony or spiral, dont have the luxury to use roadrunner. So I'd like to know if spiral is battle tested also with traditional web servers.

**Comment by butschster** [comment]
**Hi, @ardabeyazoglu** 

Sorry for long response! 

First of all. the reason of using RoadRunner is to have the best performance, because you don't need to bootstrap whole framework every request and spend only time for request handling. With `php-fpm` you will bootstrap a framework (Laravel, Symfony, Spiral) for every request and it will reduce performance. 

Without RoadRunner you won't have an ability to use all cool features, such as `jobs`, `websockets`, `metrics`, `services`, `grpc`, `cache`, `tcp server` and `fileserver`.

Here https://www.techempower.com/benchmarks/#section=data-r21&hw=ph&test=fortune&l=zik073-sf you can see benchmarks for laravel, symfony and Spiral Framework.

**Reply by ardabeyazoglu** [reply]
thanks, i will take a look
